src/Microsoft.AspNet.Mvc.ModelBinding/Validation/DataAnnotationsModelValidatorProvider.cs

```diff
@@ -19,100 +19,18 @@ namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
     /// </summary>
     public class DataAnnotationsModelValidatorProvider : IModelValidatorProvider
     {
-        // A factory for validators based on ValidationAttribute.
-        internal delegate IModelValidator DataAnnotationsModelValidationFactory(ValidationAttribute attribute);
-
-        // Factories for validation attributes
-        private static readonly DataAnnotationsModelValidationFactory _defaultAttributeFactory =
-            (attribute) => new DataAnnotationsModelValidator(attribute);
-
-        // Factories for IValidatableObject models
-        private static readonly DataAnnotationsValidatableObjectAdapterFactory _defaultValidatableFactory =
-            () => new ValidatableObjectAdapter();
-
-        private readonly Dictionary<Type, DataAnnotationsModelValidationFactory> _attributeFactories =
-            BuildAttributeFactoriesDictionary();
-
-        // A factory for validators based on IValidatableObject
-        private delegate IModelValidator DataAnnotationsValidatableObjectAdapterFactory();
-
-        internal Dictionary<Type, DataAnnotationsModelValidationFactory> AttributeFactories
-        {
-            get { return _attributeFactories; }
-        }
-
         public void GetValidators(ModelValidatorProviderContext context)
         {
             foreach (var attribute in context.ValidatorMetadata.OfType<ValidationAttribute>())
             {
-                DataAnnotationsModelValidationFactory factory;
-                if (!_attributeFactories.TryGetValue(attribute.GetType(), out factory))
-                {
-                    factory = _defaultAttributeFactory;
-                }
-
-                context.Validators.Add(factory(attribute));
+                context.Validators.Add(new DataAnnotationsModelValidator(attribute));
             }
 
             // Produce a validator if the type supports IValidatableObject
             if (typeof(IValidatableObject).IsAssignableFrom(context.ModelMetadata.ModelType))
```

Comment:
nit: strange to have `IValidatableObject` handled in this class while implementations of `IClientModelValidator` are handled in their own provider.  suggest `DataAnnotationsModelValidatorProvider` should also be made specific to `ValidationAttribute`